vcd/resource_vcd_lb_app_profile_test.go

```diff
@@ -0,0 +1,288 @@
+		if !strings.Contains(err.Error(), "could not find load balancer application profile") ||
+			monitor != nil {
+			return fmt.Errorf("load balancer application profile was not deleted: %s", err)
+		}
+		return nil
+	}
+}
+
+const testAccVcdLBAppProfile_TCP = `
+resource "vcd_lb_app_profile" "test" {
+	org          = "{{.Org}}"
+	vdc          = "{{.Vdc}}"
+	edge_gateway = "{{.EdgeGateway}}"
+  
+	name           = "{{.AppProfileName}}"
+	type           = "{{.Type}}"
+}
+
+data "vcd_lb_app_profile" "test" {
+	org          = "{{.Org}}"
+	vdc          = "{{.Vdc}}"
+	edge_gateway = "{{.EdgeGateway}}"
+	name         = "${vcd_lb_app_profile.test.name}"
+}
+`
+
+const testAccVcdLBAppProfile_UDP = `
+resource "vcd_lb_app_profile" "test" {
+	org          = "{{.Org}}"
+	vdc          = "{{.Vdc}}"
+	edge_gateway = "{{.EdgeGateway}}"
+  
+	name           = "{{.AppProfileName}}"
+	type           = "{{.Type}}"
+}
+
+data "vcd_lb_app_profile" "test" {
+	org          = "{{.Org}}"
+	vdc          = "{{.Vdc}}"
+	edge_gateway = "{{.EdgeGateway}}"
+	name         = "${vcd_lb_app_profile.test.name}"
+}
+`
+
+const testAccVcdLBAppProfile_HTTP_Cookie = `
+resource "vcd_lb_app_profile" "test" {
+  org          = "{{.Org}}"
+  vdc          = "{{.Vdc}}"
+  edge_gateway = "{{.EdgeGateway}}"
+
+  name = "{{.AppProfileName}}"
+  type = "{{.Type}}"
+
+  http_redirect_url              = "/service-one"
+  persistence_mechanism          = "cookie"
+  cookie_name                    = "JSESSIONID"
+  cookie_mode                    = "insert"
+  insert_x_forwarded_http_header = "true"
+}
+
+data "vcd_lb_app_profile" "test" {
+  org          = "{{.Org}}"
+  vdc          = "{{.Vdc}}"
+  edge_gateway = "{{.EdgeGateway}}"
+  name         = "${vcd_lb_app_profile.test.name}"
+}  
+`
+
+const testAccVcdLBAppProfile_HTTP_SourceIP = `
+resource "vcd_lb_app_profile" "test" {
+  org          = "{{.Org}}"
+  vdc          = "{{.Vdc}}"
+  edge_gateway = "{{.EdgeGateway}}"
+
+  name = "{{.AppProfileName}}"
+  type = "{{.Type}}"
+
+  http_redirect_url     = ""
+  persistence_mechanism = "sourceip"
+  # expiration = "17"
+}
+
+data "vcd_lb_app_profile" "test" {
+  org          = "{{.Org}}"
+  vdc          = "{{.Vdc}}"
+  edge_gateway = "{{.EdgeGateway}}"
+  name         = "${vcd_lb_app_profile.test.name}"
+}
+`
+
+const testAccVcdLBAppProfile_HTTPS = `
+resource "vcd_lb_app_profile" "test" {
+  org          = "{{.Org}}"
+  vdc          = "{{.Vdc}}"
+  edge_gateway = "{{.EdgeGateway}}"
+
+  name = "{{.AppProfileName}}"
+  type = "{{.Type}}"
+
+  # persistence_mechanism = "sourceip"
+  # expiration = "13"
+  enable_ssl_passthrough         = "true"
+  enable_pool_side_ssl           = "true"
+  insert_x_forwarded_http_header = "true"
+}
+
+data "vcd_lb_app_profile" "test" {
+  org          = "{{.Org}}"
+  vdc          = "{{.Vdc}}"
+  edge_gateway = "{{.EdgeGateway}}"
+  name         = "${vcd_lb_app_profile.test.name}"
+}
```

Comment:
Missing update test

Reply:
It's not. If you see there are quite a few steps and all of them except 0 are updating  because the resource name does not change.

Reply:
oh. ok. Missed that